official/keras_application_models/benchmark_main.py

```diff
@@ -0,0 +1,242 @@
+
+  # Load the model
+  tf.logging.info("Benchmark on {} model...".format(FLAGS.model))
+  keras_model = MODELS[FLAGS.model]
+  model = keras_model(weights=None)
+
+  image_size = get_default_image_size(FLAGS.model)
+  if FLAGS.use_synthetic_data:
+    tf.logging.info("Using synthetic dataset...")
+    batch_size = resnet_run_loop.per_device_batch_size(
+        FLAGS.batch_size, flags_core.get_num_gpus(FLAGS))
+    train_dataset = generate_synthetic_input_dataset(image_size, batch_size)
+    val_dataset = generate_synthetic_input_dataset(image_size, batch_size)
+  else:
+    # Use the actual ImageNet dataset (TODO)
+    pass
+
+  # If run with multiple GPUs
+  num_gpus = flags_core.get_num_gpus(FLAGS)
+  if num_gpus > 0:
+    model = tf.keras.utils.multi_gpu_model(model, gpus=num_gpus)
+
+  # Configure the model
+  model.compile(loss="categorical_crossentropy",
+                optimizer="sgd",
+                metrics=["accuracy"])
+
+  # Create benchmark logger for benchmark logging
+  run_params = {
+      "batch_size": FLAGS.batch_size,
+      "synthetic_data": FLAGS.use_synthetic_data,
+      "benchmark_level": FLAGS.benchmark_level,
+      "train_epochs": FLAGS.train_epochs
+  }
+  benchmark_logger = logger.config_benchmark_logger(FLAGS)
+  benchmark_logger.log_run_info(
+      model_name=FLAGS.model,
+      dataset_name="ImageNet",
+      run_params=run_params)
+
+  if FLAGS.use_synthetic_data:
+    train_num_images = batch_size
+    val_num_images = batch_size
+
+  # Create callbacks that log metric values about the training and evaluation
+  callbacks = get_model_callbacks(
+      FLAGS.benchmark_level, FLAGS.callbacks,
+      batch_size=FLAGS.batch_size, epoch_size=train_num_images,
+      metrics=None, metric_logger=benchmark_logger)
+
+  # Train and evaluate the model
+  history = model.fit(
+      train_dataset,
+      epochs=FLAGS.train_epochs,
+      callbacks=callbacks,
+      validation_data=val_dataset,
+      steps_per_epoch=int(np.ceil(train_num_images / float(batch_size))),
+      validation_steps=int(np.ceil(val_num_images / float(batch_size)))
+  )
+
+  tf.logging.info("Logging the evaluation results...")
+  for epoch in range(FLAGS.train_epochs):
+    eval_results = {
+        "accuracy": history.history["val_acc"][epoch],
+        "loss": history.history["val_loss"][epoch],
+        "epoch": epoch + 1,
+        tf.GraphKeys.GLOBAL_STEP: (epoch + 1) * np.ceil(
+            train_num_images/batch_size)
+    }
+    benchmark_logger.log_evaluation_result(eval_results)
+
+  # Clear the session explicitly to avoid session delete error
+  tf.keras.backend.clear_session()
+
+
+def define_keras_benchmark_flags():
+  """Add flags for keras built-in application models."""
+  flags_core.define_base()
+  flags_core.define_performance()
+  flags_core.define_image()
+  flags_core.define_benchmark()
+  flags.adopt_module_key_flags(flags_core)
+
+  flags_core.set_defaults(
+      data_format="channels_last",
```

Comment:
Why does data format have a non-standard default?

Reply:
The default is None in utils, and the models require channels_last data format.